engine/crates/engine/src/registry/resolvers/graphql/serializer.rs

```diff
@@ -469,10 +470,22 @@ impl<'a: 'b, 'b: 'a, 'c: 'a> Serializer<'a, 'b> {
         self.writeln_str("}\n")
     }
 
-    fn remove_prefix_from_type<'x>(&self, ty: &'x str) -> &'x str {
+    fn remove_prefix_from_type<'x>(&self, ty: &'x str) -> String {
         // We remove the `prefix` from condition types, as these are local to Grafbase, and
         // should not be sent to the upstream server.
-        ty.strip_prefix(self.prefix.unwrap_or_default()).unwrap_or(ty)
+        let wrappers = WrappingType::all_for(ty);
+        let ty = MetaTypeName::concrete_typename(ty);
+        let mut ty = ty
+            .strip_prefix(self.prefix.unwrap_or_default())
+            .unwrap_or(ty)
+            .to_string();
+        for wrapper in wrappers {
+            ty = match wrapper {
+                WrappingType::List => format!("[{}]", ty),
+                WrappingType::NonNull => format!("{}!", ty),
+            };
+        }
+        ty
```

Comment:
Imagine we're not doing this enough times for it to matter, but you could do this with less repeated string allocations:

```suggestion
        let wrappers = WrappingType::all_for(ty).collect::<Vec<_>>();
        let mut ty = String::with_capacity(ty.len());
        for wrapper in &wrappers {
            if let WrappingType::List = wrapper {
                write!(&mut ty, "[").ok();
            }
        }
        
        let stripped_type = MetaTypeName::concrete_typename(ty)
             .strip_prefix(self.prefix.unwrap_or_default())
            .unwrap_or(ty)
            .to_string();
        write!(&mut ty, "{stripped_type}", ).ok();
        
        for wrapper in wrappers.iter().rev() {
            match wrapper {
                WrappingType::List => write!(&mut ty, "]").ok(),
                WrappingType::NonNull => write!(&mu ty, "!").ok(),
            };
        }
        ty
```

Reply:
oh nice! Wasn't my proudest piece of code to be honest xD